homeassistant/components/template/entity.py

```diff
@@ -72,6 +86,34 @@ def referenced_blueprint(self) -> str | None:
     def _render_script_variables(self) -> dict:
         """Render configured variables."""
 
+    @abstractmethod
+    def setup_state_template(
+        self,
+        option: str,
+        attribute: str,
+        validator: Callable[[Any], Any] | None = None,
+        on_update: Callable[[Any], None] | None = None,
+    ) -> None:
+        """Set up a template that manages the main state of the entity."""
+
+    def add_template(
+        self,
+        option: str,
+        attribute: str,
+        validator: Callable[[Any], Any] | None = None,
+        on_update: Callable[[Any], None] | None = None,
+        none_on_template_error: bool = False,
+    ) -> Template | None:
+        """Add a template."""
+        if (template := self._config.get(option)) and isinstance(template, Template):
+            if not template.is_static:
+                self._templates[option] = EntityTemplate(
+                    attribute, template, validator, on_update, none_on_template_error
+                )
+            return template
+
+        return None
```

Comment:
It's not clear to me how this method relates to `add_template_attribute`, can you explain it?